File: ojuz/ioi/2014/Day1/game-templates/grader.pas
program grader;

uses game;

var
    n, u, v, i: longint;

function read_int(): longint;
var
    x: longint;
begin
    read(x);
    read_int := x;
end;


begin
    n := read_int();
    initialize(n);
    for i := 1 to n * (n - 1) div 2 do
    begin
        u := read_int();
        v := read_int();
        writeln(hasEdge(u, v));
    end;
end.

File: ojuz/ioi/2014/Day1/game-templates/grader.cpp
#include <cstdio>
#include <cassert>
#include "game.h"

int read_int() {
    int x;
    assert(scanf("%d", &x) == 1);
    return x;
}

int main() {
    int n, u, v;
    n = read_int();
    initialize(n);
    for (int i = 0; i < n * (n - 1) / 2; i++) {
        u = read_int();
        v = read_int();
        printf("%d\n", hasEdge(u, v));
    }
    return 0;
}
